packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -681,31 +694,42 @@ class AppInstance {
 
 /// A [Logger] which sends log messages to a listening daemon client.
 class _AppRunLogger extends Logger {
-  _AppRunLogger(this.domain, this.app);
+  _AppRunLogger(this.domain, this.app, { this.logToStdout: false });
 
   AppDomain domain;
   final AppInstance app;
+  final bool logToStdout;
   int _nextProgressId = 0;
 
   @override
   void printError(String message, [StackTrace stackTrace]) {
-    if (stackTrace != null) {
-      _sendLogEvent(<String, dynamic>{
-        'log': message,
-        'stackTrace': stackTrace.toString(),
-        'error': true
-      });
+    if (logToStdout) {
+      stderr.writeln(message);
+      if (stackTrace != null)
+        stderr.writeln(stackTrace.toString().trimRight());
     } else {
-      _sendLogEvent(<String, dynamic>{
-        'log': message,
-        'error': true
-      });
+      if (stackTrace != null) {
+        _sendLogEvent(<String, dynamic>{
+          'log': message,
+          'stackTrace': stackTrace.toString(),
+          'error': true
+        });
+      } else {
+        _sendLogEvent(<String, dynamic>{
+          'log': message,
+          'error': true
+        });
+      }
     }
   }
 
   @override
   void printStatus(String message, { bool emphasis: false, bool newline: true, String ansiAlternative }) {
-    _sendLogEvent(<String, dynamic>{ 'log': message });
+    if (logToStdout) {
+      print(message);
```

Comment:
Similar to my earlier comment about `print(...` and `stderr.writeln(...`, should this statement be `stdout.writeln(...`?